### PR TITLE
test: add test for unint64 overflow on limit

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

Wrap around is guaranteed in this case, ref https://go.dev/ref/spec#Integer_overflow. For signed ints this is also deterministic apparently.

closes: #XXXX

### Commit Message / Changelog Entry

```text
test: add test that asserts values after limit wraps around.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/keeper_test.go [hunk 1]
```diff
@@ -811,6 +812,28 @@ func (suite *KeeperTestSuite) TestPruneAcknowledgements() {
...
+				suite.sendMockPackets(path, 10, true)
+			},
+			func() {
+				limit = math.MaxUint64
+			},
+			func(pruned, left uint64) {
+				// Nothing should be pruned, by passing in a limit of math.MaxUint64, overflow occurs
+				// when initializing end to pruningSequenceStart + limit. This results in end always being
+				// equal to start - 1 and thereby not entering the for loop.
+				// We expect 10 acks, 10 receipts and pruningSequenceStart == 1 (loop not entered).
```
Comment (colin-axner) [inline modules/core/04-channel/keeper/keeper_test.go:828]: crazy, so it doesn't panic, it just wraps around. I guess by construction this is safe. Should we add an in-line comment in the code in case someone tries to refactor it?

**Review by colin-axner — APPROVED** [review]
Thank you @DimitrisJim !!!

**Review by crodriguezvega — APPROVED** [review]
Thanks for checking this!

**Comment by damiannolan** [comment]
Awesome @DimitrisJim, nice one!
